openapi_python_client/parser/openapi.py

```diff
@@ -49,6 +52,8 @@ def from_data(
                     endpoint, schemas = Endpoint._add_parameters(
                         endpoint=endpoint, data=path_data, schemas=schemas, config=config
                     )
+                if not isinstance(endpoint, ParseError):
+                    endpoint = Endpoint._sort_parameters(endpoint=endpoint, path=path)
```

Comment:
Note that is must be outside the `_add_parameters` function (as it is called twice, first for method parameters and then for the path parameters that affect all methods).

So while writing this code I noticed that renaming of duplicate name parameters should be done outside `_add_parameters` function also, as it not manages the case that a parameter is defined outside of method specific parameter, resulting in ParseError.

Reply:
That's fixed in the upcoming #457 🎉 !